internal/clients/substrate/substrate.go

```diff
@@ -0,0 +1,74 @@
+package substrate
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/threefoldtech/tf-kyc-verifier/internal/logger"
+
+	// use tfchain go client
+
+	tfchain "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+)
+
+type Substrate struct {
+	api    *tfchain.Substrate
+	logger logger.Logger
+}
+
+func New(config SubstrateConfig, logger logger.Logger) (*Substrate, error) {
```

Comment:
There is no need to pass configuration interface here.

Reply:
Wouldn't it be cleaner to use the interface instead of depending on config models?

Reply:
No, As long as you don't have more than one implementation, introducing an abstraction that you don't need right now and possible won't need at all defeats the purpose of KISS/YAGNI